cmd/main.go

```diff
@@ -122,8 +119,12 @@ func main() {
 		go pprofStartServer(flagVar.PprofAddr, flagVar.PprofServerTimeout, setupLog)
 	}
 
-	cacheOptions := internal.GetCacheOptions(flagVar.IsKymaManaged, flagVar.IstioNamespace,
-		flagVar.IstioGatewayNamespace)
+	cacheOptions := setup.SetupCacheOptions(flagVar.IsKymaManaged,
```

Comment:
Proposal to not have the full setup in main (see also below `skrWebhookManager = setup.SetupSkrWebhookManager(mgr, skrContextProvider, flagVar, logger)`) 

However, a concrete decision how we will go forward with this will be taken in https://github.com/kyma-project/lifecycle-manager/pull/2395

Reply:
nice, this is way better than before👍  I'm still thinking about how to handle the dependency compliance with the single-level abstraction principle. Basically it comes back to the previous discussion: if we treat CertificateManager as internal component of SKRWebhookManifestManager, is it necessary to expose it to other level only for construction, which violate the SLA.

Let's keep your solution, I will propose some options in the meeting for us to discuss.